### ISSUE TITLE
[PVR] Removing stale channels does not use bulk SQL commits

### ISSUE THREAD

**Issue by basilgello** [xbmc/xbmc#18676]
### Describe the bug

The removal of stale PVR channels takes long time and excessive HDD usage similar to #17800

## Expected Behavior

Removal of stale channels should be bulk just like it is now on update.

## Actual Behavior

Epg/TV database journals got creared and removed, GDB breakpoint on `open` yields the following stacktrace:

```
Thread 25 "PVRManager" hit Breakpoint 1, 0x00007fa88086cc40 in open64 () from /lib/x86_64-linux-gnu/libpthread.so.0
(gdb) bt
#0  0x00007fa88086cc40 in open64 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007fa87cfb8e6e in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#2  0x00007fa87cfb9068 in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#3  0x00007fa87cfb98fc in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#4  0x00007fa87cfe8c94 in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#5  0x00007fa87cfe924c in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#6  0x00007fa87cfe947f in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#7  0x00007fa87cfedff7 in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#8  0x00007fa87d011f76 in ?? () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#9  0x00007fa87d012ada in sqlite3_step () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#10 0x00007fa87d0139a5 in sqlite3_exec () from /lib/x86_64-linux-gnu/libsqlite3.so.0
#11 0x0000564278fa35b5 in dbiplus::SqliteDataset::exec(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#12 0x0000564278f945dd in CDatabase::ExecuteQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#13 0x0000564278f946fe in CDatabase::DeleteValues(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CDatabase::Filter const&) ()
#14 0x00005642792ffb1d in PVR::CPVRDatabase::Delete(PVR::CPVRChannel const&) ()
#15 0x00005642792c09dc in PVR::CPVRChannel::Delete() ()
#16 0x00005642792cda08 in PVR::CPVRChannelGroupInternal::RemoveDeletedChannels(PVR::CPVRChannelGroup const&) ()
#17 0x00005642792c773d in PVR::CPVRChannelGroup::UpdateGroupEntries(PVR::CPVRChannelGroup const&, std::vector<std::shared_ptr<PVR::CPVRChannel>, std::allocator<std::shared_ptr<PVR::CPVRChannel> > >&) ()
#18 0x00005642792cda4e in PVR::CPVRChannelGroupInternal::UpdateGroupEntries(PVR::CPVRChannelGroup const&, std::vector<std::shared_ptr<PVR::CPVRChannel>, std::allocator<std::shared_ptr<PVR::CPVRChannel> > >&) ()
#19 0x00005642792ccea2 in PVR::CPVRChannelGroupInternal::Update(std::vector<std::shared_ptr<PVR::CPVRChannel>, std::allocator<std::shared_ptr<PVR::CPVRChannel> > >&) ()
#20 0x00005642792c7f32 in PVR::CPVRChannelGroup::Load(std::vector<std::shared_ptr<PVR::CPVRChannel>, std::allocator<std::shared_ptr<PVR::CPVRChannel> > >&) ()
#21 0x00005642792cd7b4 in PVR::CPVRChannelGroupInternal::Load(std::vector<std::shared_ptr<PVR::CPVRChannel>, std::allocator<std::shared_ptr<PVR::CPVRChannel> > >&) ()
#22 0x00005642792d0f89 in PVR::CPVRChannelGroups::Load() ()
#23 0x00005642792d25c3 in PVR::CPVRChannelGroupsContainer::Load() ()
#24 0x0000564279308e43 in PVR::CPVRManager::LoadComponents(PVR::CPVRGUIProgressHandler*) ()
#25 0x000056427930900e in PVR::CPVRManager::Process() ()
#26 0x0000564278e0df6d in CThread::Action() ()
#27 0x0000564278e0f401 in ?? ()
#28 0x0000564278e0fc9a in ?? ()
#29 0x00007fa87c13eb2f in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#30 0x00007fa880862fa3 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#31 0x00007fa87c3004cf in clone () from /lib/x86_64-linux-gnu/libc.so.6
```

### To Reproduce

1. Start Kodi with pvr.iptvsimple
2. Let it pick updated M3U with thousand channels removed from playlist
3. Wait 10min

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian 10 Buster
 - Kodi version: Kodi master

**Comment by basilgello** [comment]
@phunkyfish @ksooo

**Comment by ksooo** [comment]
PR welcome. ;-)

**Comment by ksooo** [comment]
@basilgello fixed by #18700
